src/CashuWallet.ts

```diff
@@ -365,26 +626,44 @@ class CashuWallet {
 	}
 
 	/**
-	 * Mint tokens for a given mint quote
+	 * Mint proofs for a given mint quote
 	 * @param amount amount to request
 	 * @param quote ID of mint quote
+	 * @param options.keysetId? optionally set keysetId for blank outputs for returned change.
+	 * @param preference? Deprecated. Use `outputAmounts` instead. Optional preference for splitting proofs into specific amounts.
+	 * @param outputAmounts? optionally specify the output's amounts to keep and to send.
+	 * @param counter? optionally set counter to derive secret deterministically. CashuWallet class must be initialized with seed phrase to take effect
+	 * @param pubkey? optionally locks ecash to pubkey. Will not be deterministic, even if counter is set!
 	 * @returns proofs
 	 */
-	async mintTokens(
+	async mintProofs(
 		amount: number,
 		quote: string,
 		options?: {
 			keysetId?: string;
-			preference?: Array<AmountPreference>;
+			outputAmounts?: OutputAmounts;
+			proofsWeHave?: Array<Proof>;
 			counter?: number;
 			pubkey?: string;
 		}
 	): Promise<{ proofs: Array<Proof> }> {
 		const keyset = await this.getKeys(options?.keysetId);
+		if (!options?.outputAmounts && options?.proofsWeHave) {
+			options.outputAmounts = {
```

Comment:
`options` should be defined, due to `&& options?.proofsWeHave` conditional. Mutating state outside of the function is real though, which I agree should be avoided. In this case though, it might not be as bad, since it will update the state to what it should have been in the first place.

If you got a good idea for a cleaner solution, feel free to add it to the PR

Reply:
Yep, you are right. Looking at this again this assignment might even be pretty safe, as it only happens when `outputAmounts` is falsy.